source/declarations.tex

```diff
@@ -816,78 +810,6 @@
 \end{codeblock}
 \end{example}
 
-\pnum
-\indextext{specifier!\idxcode{constexpr}!constructor}%
-The definition of a constexpr constructor
-whose \grammarterm{function-body} is not \tcode{= delete}
-shall additionally satisfy the following requirements:
-\begin{itemize}
-
-\item
-for a non-delegating constructor, every constructor selected to initialize non-static
-data members and base class subobjects shall be a constexpr constructor;
-
-\item
-for a delegating constructor, the target constructor shall be a constexpr
-constructor.
-\end{itemize}
-
-\begin{example}
-\begin{codeblock}
-struct Length {
-  constexpr explicit Length(int i = 0) : val(i) { }
-private:
-  int val;
-};
-\end{codeblock}
-\end{example}
-
-\pnum
-The definition of a constexpr destructor
-whose \grammarterm{function-body} is not \tcode{= delete}
-shall additionally satisfy the following requirement:
-\begin{itemize}
-\item
-  for every subobject of class type or
-  (possibly multi-dimensional) array thereof,
-  that class type shall have a constexpr destructor.
-\end{itemize}
-
-\pnum
-For a constexpr function or constexpr constructor
-that is neither defaulted nor a template,
-if no argument values exist such that
-an invocation of the function or constructor could be an evaluated subexpression of a core
-constant expression\iref{expr.const}, or,
-for a constructor, an evaluated subexpression of
-the initialization full-expression of some constant-initialized object\iref{basic.start.static},
-the program is ill-formed, no diagnostic required.
-\begin{example}
-\begin{codeblock}
-constexpr int f(bool b)
-  { return b ? throw 0 : 0; }           // OK
-constexpr int f() { return f(true); }   // ill-formed, no diagnostic required
-
-struct B {
-  constexpr B(int x) : i(0) { }         // \tcode{x} is unused
-  int i;
-};
-
-int global;
-
-struct D : B {
-  constexpr D() : B(global) { }         // ill-formed, no diagnostic required
-                                        // lvalue-to-rvalue conversion on non-constant \tcode{global}
-};
-
-constexpr int f(int x) {
-  static int n = x;
-  return n + x;                         // ill-formed, no diagnostic required
-                                        // all calls reach the static variable declaration
```

Comment:
This change isn't in the paper (which I suppose has a stale baseline). This example was added by https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2021/p2242r3.html.

@jensmaurer Is it OK to lose this example?

Reply:
I think we want to keep that example. Can you revert that hunk?

Reply:
We've lost the ambient paragraph, though, so where should the residual example be? Just in a paragraph of its own?

Reply:
But the paper normatively removes the restriction that it's IFNDR, so the example no longer has normative backing. If it does have, the example can be moved to where that is justified.

Reply:
I see, thanks. @jensmaurer OK to lose this then?

Reply:
Right, ok with keeping the example removed.

Reply:
Great, thank you both!